frontend/src/pages/Interventions/FourthActivity.tsx

```diff
@@ -68,34 +77,48 @@ const ButtonContainer = styled.div`
   justify-content: end;
 `;
 
+const connector = connect(
+  (state) => ({
+    interventions: getCurrentInterventions(state),
+  }),
+  {
+    updateIntervention: updateIntervention.request,
+  }
+);
+
 const FourthActivity = ({
   title,
   imageUrl,
+  interventions,
+  updateIntervention,
 }: FourthActivityProps): ReactElement => {
+  const activityIdx = interventions && interventions.activityIdx;
+  const wordIdx = interventions && interventions.wordIdx;
+  const nextActivityIdx = getNextActivityIdx(activityIdx);
+  // const nextWordIdx = getNextWordIdx(wordIdx);
```

Comment:
This seems to be the same between Fourth and FirstActivity. I wonder if this should be passed in from the parent. Like I wonder if these should not worry about how to get data, but more just take the data and use it?

Reply:
ohh got point i see what u mean!

Reply:
oh wait actually since the activity and word idx are changing, they need to be in the component instead of in the parent correct me if im wrong tho